tests/components/kraken/test_config_flow.py

```diff
@@ -0,0 +1,121 @@
+
+from .const import TICKER_INFORMATION_RESPONSE, TRADEABLE_ASSET_PAIR_RESPONSE
+
+from tests.common import MockConfigEntry
+
+
+def init_config_flow(hass):
+    """Init a configuration flow."""
+    flow = config_flow.KrakenConfigFlow()
+    flow.hass = hass
+    return flow
+
+
+async def test_config_flow(hass):
+    """Test we can finish a config flow."""
+    with patch(
+        "pykrakenapi.KrakenAPI.get_tradable_asset_pairs",
+        return_value=TRADEABLE_ASSET_PAIR_RESPONSE,
+    ):
+        with patch(
+            "pykrakenapi.KrakenAPI.get_ticker_information",
+            return_value=TICKER_INFORMATION_RESPONSE,
+        ):
+            result = await hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": "user"}
+            )
+            assert result["type"] == "form"
+
+            result = await hass.config_entries.flow.async_configure(
+                result["flow_id"], {"name": "test"}
+            )
+            assert result["type"] == "create_entry"
+
+            await hass.async_block_till_done()
+            state = hass.states.get("sensor.test_ada_eth_ask")
+            assert state
+
+
+async def test_form_already_configured(hass):
+    """Test is already configured."""
+    with patch(
+        "pykrakenapi.KrakenAPI.get_tradable_asset_pairs",
+        return_value=TRADEABLE_ASSET_PAIR_RESPONSE,
+    ):
+        with patch(
+            "pykrakenapi.KrakenAPI.get_ticker_information",
+            return_value=TICKER_INFORMATION_RESPONSE,
+        ):
+            result = await hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": "user"}
+            )
+            assert result["type"] == "form"
+
+            result = await hass.config_entries.flow.async_configure(
+                result["flow_id"], {"name": "test"}
+            )
+            assert result["type"] == "create_entry"
+
+            await hass.async_block_till_done()
+
+            result = await hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": "user"}
+            )
+            assert result["type"] == "form"
+
+            result = await hass.config_entries.flow.async_configure(
+                result["flow_id"], {"name": "test"}
+            )
+
+            assert result["type"] == "abort"
+            assert result["reason"] == "already_configured"
+
+
+async def test_options(hass):
+    """Test options for Kraken."""
+    with patch(
+        "pykrakenapi.KrakenAPI.get_tradable_asset_pairs",
+        return_value=TRADEABLE_ASSET_PAIR_RESPONSE,
+    ):
+        with patch(
+            "pykrakenapi.KrakenAPI.get_ticker_information",
+            return_value=TICKER_INFORMATION_RESPONSE,
+        ):
+            entry = MockConfigEntry(
+                domain=DOMAIN, data={"name": "kraken"}, options={CONF_SCAN_INTERVAL: 60}
+            )
+            entry.add_to_hass(hass)
+            flow = init_config_flow(hass)
```

Comment:
Please always use the core interface to start the tests. Here use `hass.config_entries.options.async_init`.

Reply:
Done